File: DownloadFiles/TGNet/Chapter05/C#/Exercise5-2/AutoTesterwithStaticClass/AutoTester/TestLib.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Diagnostics;
using System.IO;


    static class TestLib
    {
        public static string Logfile = "c:\\logfile.txt";
        
        public static void StartProgram(string strAppName)
        {
            if (!(File.Exists(Logfile)))
            {
                CreateLog("Starting Log file " + DateTime.Now);
            }
            Process.Start(strAppName);
            LogtoFile("Application " + strAppName + " Started at: " + DateTime.Now);
        }

        private static void CreateLog(string strHeader)
        {
            StreamWriter Writer;
            Writer = File.CreateText(Logfile);
            Writer.WriteLine(strHeader);
            Writer.Close();
        }

        private static void LogtoFile(string strLog)
        {
            StreamWriter Writer;
            Writer = File.AppendText(Logfile);
            Writer.WriteLine(strLog);
            Writer.Close();
        }
    }

File: DownloadFiles/TGNet/Chapter05/VB/Exercise5-2/AutoTesterwithSharedClass/Chapter5Demo/TestLib.vb
Imports System.Diagnostics
Imports System.IO

Public Class TestLib
    Public Shared Logfile As String = "c:\logfile.txt"

    Public Shared Sub StartProgram(ByVal strAppname As String)
        'This routine will start a new process 
        'It will log the starting time 

        If Not File.Exists(Logfile) Then
            CreateLog("Starting Log file " & DateTime.Now)
        End If
        Process.Start(strAppname)
        LogtoFile("Application " & strAppname & " Started at: " & DateTime.Now)
    End Sub
    Public Shared Sub CreateLog(ByVal strHeader As String)
        Dim Writer As StreamWriter
        Writer = File.CreateText(Logfile)
        Writer.WriteLine(strHeader)
        Writer.Close()
    End Sub
    Public Shared Sub LogtoFile(ByVal strLog As String)
        Dim Writer As StreamWriter
        Writer = File.AppendText(Logfile)
        Writer.WriteLine(strLog)
        Writer.Close()
    End Sub


End Class

